### PR TITLE
Make "Publish to Google Cloud..." remember previous settings.

### LINKED ISSUE #863: Feature request: One-click re-publish
When using the VS plugin to "Publish to Google Cloud..." it would be very helpful if the publication parameters could be auto-filled from a previous publish operation.

So, for example, having published once to GKE, I can re-publish without having to re-select GKE, and re-select the cluster, deployment name, replica count, etc...

### PR DESCRIPTION
**Pull request by ILMTitan**
Fixes #863.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialog/Steps/Choice/Choice.cs [hunk 1]
```diff
@@ -23,24 +23,38 @@ namespace GoogleCloudExtension.PublishDialog.Steps.Choice
     /// </summary>
     public class Choice : Model
     {
+        /// <summary>
+        /// Gets the ID string of the choice. i.e. Gae, Gce, Gke.
+        /// </summary>
+        public string Id { get; }
```
Comment (redborian) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialog/Steps/Choice/Choice.cs]: "EnvType" would be better

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialog/Steps/Gke/GkeStepViewModel.cs [hunk 2]
```diff
@@ -331,13 +388,16 @@ protected internal override void OnFlowFinished()
                         DeploymentName = DeploymentName,
                         DeploymentVersion = DeploymentVersion,
                         ExposeService = ExposeService,
-                        ExposePublicService = ExposePublicService,
+                        ExposePublicService = ExposeService && ExposePublicService,
```
Comment (redborian) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialog/Steps/Gke/GkeStepViewModel.cs]: Looks like exposing service and exposing public service are two separate checkboxes. Maybe let the functionality as is?

**Comment by iantalarico** [comment]
+1 to what Chandra said here.  This is a massive PR.  Can you break this up into some chucks to make the review easier?

**Comment by ILMTitan** [comment]
Closing this to create separate PRs.

**Pull request closed without merging**